charts/aspnetcore/templates/deployment.yaml

```diff
@@ -89,12 +89,16 @@ spec:
       {{- end }}
       volumes:
       {{- if .Values.certificateStore.enabled }}
-      - name: internal-certificates
+      - name: certificate-store
         csi:
-          driver: secrets-store.csi.k8s.io
-          readOnly: true
+          driver: file.csi.azure.com
           volumeAttributes:
-            secretProviderClass: internal-dns-keyvault
+            subscriptionID: {{ quote .Values.certificateStore.subscriptionId }}
+            resourceGroup: {{ quote .Values.certificateStore.resourceGroup }}
+            storageAccount: {{ quote .Values.certificateStore.storageAccount }}
+            shareName: {{ quote .Values.certificateStore.shareName }}
+            clientID: {{ quote .Values.azureWorkloadIdentity.clientId }}
+            mountOptions: "dir_mode=0777,file_mode=0777,cache=strict,actimeo=30,nosharesock"
```

Comment:
These options come from the example [here](https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/master/deploy/example/nginx-pod-azurefile-inline-volume.yaml). Even though this is mounted with 777 permissions, the share is read-only and nothing can be written to it.